### PR TITLE
docs: Document how to run SQLFluff with local changes to test them

### LINKED ISSUE #2313: Add example to parse one file in test/fixtures/dialects/...
## URL or Command Line Option with the issue being reported
https://github.com/sqlfluff/sqlfluff/blob/main/CONTRIBUTING.md

## Issue/Suggested Improvement
Would be nice to have an example in the CONTRIBUTING.md on how to parse a single file of the test fixture sql. This would make my life a little bit easier as it is much faster than to generate all test fixtures and check the corresponding yml file.

I have found a description on how to run a single test case but I have no clue on how to parse an sql file if I have changed Sqlfluff and want to validate my changes against some fixtures.

### PR DESCRIPTION
**Pull request by kayman-mk**
### Brief summary of the change made

Add a paragraph to describe on how to run SQLFluff if changed locally.

Closes #2313 

### Are there any other side effects of this change that we should be aware of?
No

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -168,6 +168,12 @@ tox -e py38 -- test/cli
 tox -e py38 -- test/cli/commands_test.py
 ```
 
+Running your code changes against a SQL file you can run
+```shell
+sqlfluff parse my-sql.sql
```
Comment (jpy-git) [inline CONTRIBUTING.md:173]:
```suggestion
sqlfluff parse test.sql
```
use `test.sql` for random tests. It's already in the `.gitignore` so people won't accidently commit them.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -168,6 +168,12 @@ tox -e py38 -- test/cli
 tox -e py38 -- test/cli/commands_test.py
 ```
 
+Running your code changes against a SQL file you can run
```
Comment (jpy-git) [inline CONTRIBUTING.md:171]: You can also manually test your updated code against a SQL file via:

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -168,6 +168,12 @@ tox -e py38 -- test/cli
...
+```shell
+sqlfluff parse my-sql.sql
+```
+in the virtual environment. This executes the locally changed SQLFluff program.
```
Comment (jpy-git) [inline CONTRIBUTING.md:175]:
```suggestion
(ensure your virtual environment is activated first)
```

**Comment by kayman-mk** [comment]
Much better now

**Review by jpy-git — APPROVED** [review]
All good thanks 🙇

**Pull request merged**